### PR TITLE
👀No statics in api client

### LINKED ISSUE #195: Ditch static class method pattern in javascript http client
We are consistently using a the following pattern for XHR service classes:

```js
// foo.service.js
class FooService {
  static fetch() {
    return ApiService.get('/foo').then(...);
  }
}
```

As discussed in peer reviews and pairing sessions, we feel a pattern of base class inheritance (`ApiService`), constructor injection and instance methods would be better for our team instead of static class methods.

The class inheritance pattern is more familiar to ruby devs, _and_ we get away from some singleton pattern code smells.

Fortunately (b/c we have been so consistent) it should be a straight-forward refactor.

### PR DESCRIPTION
**Pull request by eckdanny-osi**
Closes #195

### CODE REVIEW

**Comment by SashaKyz** [comment]
Can one of the admins verify this patch?

File: app/javascript/_services/address/address.service.js [hunk 1]
```diff
@@ -1,8 +1,10 @@
-import ApiService from '../api';
+import BaseHttpService from '../http-service';
 
-class AddressService {
-  static fetch(id) {
-    return ApiService.get(`/addresses/${id}`).then(response => response.data);
+class AddressService extends BaseHttpService {
+  static PREFIX = '/addresses';
+  fetch(id) {
+    // console.log(this._client);
```
Comment (RamuKammagani-osi) [inline app/javascript/_services/address/address.service.js]: Remove comments!

File: app/javascript/_services/address/address.service.test.js [hunk 2]
```diff
@@ -1,25 +1,26 @@
...
   });
 
   describe('#fetch', () => {
-    let getSpy;
-
-    beforeEach(() => {
-      getSpy = jest.spyOn(ApiService, 'get');
-    });
+    it('calls axios', () => {
+      console.log(axios);
```
Comment (RamuKammagani-osi) [inline app/javascript/_services/address/address.service.test.js]: same here

File: app/javascript/_services/address/address.service.test.js [hunk 3]
```diff
@@ -1,25 +1,26 @@
...
 
-    it('calls ApiService', () => {
-      getSpy.mockReturnValue(Promise.resolve({}));
-      expect(getSpy).not.toHaveBeenCalled();
-      AddressService.fetch();
-      expect(getSpy).toHaveBeenCalledWith(jasmine.any(String));
+      jest
+        .spyOn(axios, 'request')
+        .mockReturnValueOnce(Promise.resolve({ data: [] }));
+      expect(axios.get).not.toHaveBeenCalled();
```
Comment (RamuKammagani-osi) [inline app/javascript/_services/address/address.service.test.js]: Tests Fails here! just a normal fix i guess

**Comment by RamuKammagani-osi** [comment]
Oh! just noticed that its still in WIP, Sorry!

**Pull request closed without merging**